engine/src/main/java/org/terasology/engine/modes/loadProcesses/RegisterMods.java

```diff
@@ -87,8 +78,12 @@ public boolean step() {
                 }
 
                 EnvironmentSwitchHandler environmentSwitchHandler = context.get(EnvironmentSwitchHandler.class);
-                applyModulesThread = new Thread(() -> environmentSwitchHandler.handleSwitchToGameEnvironment(context));
-                applyModulesThread.start();
+
+                applyModulesThread = Executors.newSingleThreadExecutor(
```

Comment:
suggestion: Reuse same executor or shutdown executor on done or extends `ThreadManager` to return `Future`